### PR TITLE
Azure-Bake Build Update

### PR DESCRIPTION
**Pull request by csperbeck**
Updated the build to GULP for additional integration and conditional logic along with parameters, and environment variables.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    if (branchName !== 'master') {        
+        branchName = branchName.replace(/refs\/heads\/(feature\/)?/i, '');
+    }    
+    var gitScript = `sudo git checkout ${branchName}`;
+    console.log('ADO Prep Script: ' + gitScript);
+    return runCmd(gitScript, done);    
+}
+
+function bumpVersion() {
+    return gulp.src(['ingredient/**/package.json','core/package.json','system/package.json','lerna.json'], { base: './' })
```
Comment (whilke) [inline gulpfile.js]: Do we want gulp to bump versions? Doesn't lerna do that for us?

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+}
+
+function lernaPublish(done) {    
+    var gitScript = `sudo npm run publish` ;
+    console.log('Build Script: ' + gitScript);
+    return runCmd(gitScript, done);    
+}
+
+function systemPublish(done) {
+    var gitScript = `sudo npm publish --prefix ./system &&  npm run release-build --prefix ./system` ;
```
Comment (whilke) [inline gulpfile.js]: think we only want release-build here, lerna publish above takes care of all the npm publishing

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        break;        
+    }
+}
+
+function cleanCoverage() {
+    return del('coverage/**', { force: true });
+}
+
+
+function gitCommit(done) {
```
Comment (whilke) [inline gulpfile.js]: lerna already creates the chore update/publish commit, and does the push. Not sure we need this?

File: gulpfile.js (hunk 1 above)
Comment (csperbeck) [reply gulpfile.js]: bump version was for circumstances outside of lerna/npm bumping versions. AKA if a build borks and we need to bump the version outside of modifying and committing new versions.

Comment (csperbeck) [reply gulpfile.js]: removed.

**Pull request merged**